probe-rs/src/debug/variable_cache.rs

```diff
@@ -29,60 +32,116 @@ impl Serialize for VariableCache {
+            parent_variable_key: ObjectRef,
         ) -> Vec<VariableTreeNode> {
             variable_cache
                 .get_children(parent_variable_key)
                 .unwrap()
-                .iter()
-                .map(|child_variable: &Variable| VariableTreeNode {
-                    name: child_variable.name.clone(),
-                    type_name: child_variable.type_name.clone(),
-                    value: if child_variable.range_upper_bound > 50 {
+                .into_iter()
+                .map(|child_variable: Variable| {
+                    let value = if child_variable.range_upper_bound > 50 {
                         format!("Data types with more than 50 members are excluded from this output. This variable has {} child members.", child_variable.range_upper_bound)
                     } else {
                         child_variable.get_value(variable_cache)
-                    },
-                    children: if child_variable.range_upper_bound > 50 {
-                        // Empty Vec's will show as variables with no children.
-                        Vec::new()
-                    } else {
-                        recurse_variables(variable_cache, Some(child_variable.variable_key))
-                    },
-                                    })
+                    };
+
+                    VariableTreeNode {
+                                    name: child_variable.name,
+                                    type_name: child_variable.type_name,
+                                    value ,
+                                    children: if child_variable.range_upper_bound > 50 {
+                                        // Empty Vec's will show as variables with no children.
+                                        Vec::new()
+                                    } else {
+                                        recurse_variables(variable_cache, child_variable.variable_key)
+                                    },
+                                                    }
+                })
                 .collect::<Vec<VariableTreeNode>>()
         }
 
         let mut state = serializer.serialize_struct("Variables", 1)?;
-        state.serialize_field("Child Variables", &recurse_variables(self, None))?;
+        state.serialize_field("Child Variables", &recurse_cache(self))?;
         state.end()
     }
 }
 
-impl Default for VariableCache {
-    fn default() -> Self {
-        Self::new()
-    }
-}
-
 impl VariableCache {
-    /// Creates a new [`VariableCache`].
-    pub fn new() -> Self {
+    fn new(mut variable: Variable) -> Self {
+        let key = get_object_reference();
+
+        variable.variable_key = key;
+
         VariableCache {
-            variable_hash_map: HashMap::new(),
+            root_variable_key: key,
+            variable_hash_map: HashMap::from([(key, variable)]),
         }
     }
 
+    /// Create a cache for static variables for the given unit
```

Comment:
This doc comment does not match the one of the following function even tho they do the same stuff for locals & statics respectively. I would love it they used the same words to avoid confusion :) Is a Unit aequivalent to a DIE?

Reply:
I've combined the two functions, and added more docs. The only difference between them was the `VariableName`.